test/precompile.jl

```diff
@@ -2145,6 +2145,29 @@ precompile_test_harness("No backedge precompile") do load_path
     end
 end
 
+precompile_test_harness("Pre-compile Core methods") do load_path
+    # Core methods should support pre-compilation as external CI's like anything else
+    # https://github.com/JuliaLang/julia/issues/58497
+    write(joinpath(load_path, "CorePrecompilation.jl"),
+          """
+          module CorePrecompilation
+          struct Foo end
+          precompile(Tuple{Type{Vector{Foo}}, UndefInitializer, Tuple{Int}})
+          end
+          """)
+    ji, ofile = Base.compilecache(Base.PkgId("CorePrecompilation"))
+    @eval using CorePrecompilation
+    invokelatest() do
+        let tt = Tuple{Type{Vector{CorePrecompilation.Foo}}, UndefInitializer, Tuple{Int}},
+            match = first(Base._methods_by_ftype(tt, -1, Base.get_world_counter())),
+            mi = Base.specialize_method(match)
```

Comment:
```suggestion
            mi = first(Base.specializations(match.method))
```
to be certain that you're not *generating* that `mi` with this call.

Reply:
This method is very general though, it has many specializations:
```julia
julia> match.method
Vector{T}(::UndefInitializer, d::Tuple{Int64}) where T
     @ Core boot.jl:660

julia> length(Base.specializations(match.method))
108
```

I think the `cache` checks should guarantee that we're picking up an existing MethodInstance (the test fails as expected on current master)